### PR TITLE
Fix/bugs

### PR DESCRIPTION
**Pull request by manish591**
This is a refactor branch: 
- Updated categories on the home page, users can click on one of the categories and will navigate to the listing page with that selected category

- User can't navigate back to the previous route once redirected to the login route. This is because of protected routes. I'm redirecting the user to the login page if they try to navigate to a protected route without authentication. Once a user is navigating to a protected route, the route gets stored in history and on clicking the back button again we are on the protected route so it's redirecting the user back to the login route again
The solution is that once redirected from the protected route, We should remove the protected route from route history.

- Some minor changes

### CODE REVIEW

File: src/components/sidebar/MobileTopSidebar.css [hunk 1]
```diff
@@ -34,3 +40,11 @@
...
   font-size: 1.8rem;
 }
+
+.mt-sidebar-btm__list {
+  font-size: 1.1rem;
+}
+
+:target {
+  display: grid;
+}
```
Comment (varnita619) [inline src/components/sidebar/MobileTopSidebar.css:50]: you can add new line at end of each file for better formatting

File: src/pages/Home/Home.css [hunk 2]
```diff
@@ -129,6 +130,49 @@
   justify-content: center;
 }
 
+/* ---------- */
+/* categories */
+/* ---------- */
```
Comment (varnita619) [inline src/pages/Home/Home.css:135]: good use of comments

**Review by varnita619 — APPROVED** [review]
Code looks clean
UI is good
All files are structured properly

File: src/components/sidebar/MobileTopSidebar.css (hunk 1 above)
Comment (manish591) [reply src/components/sidebar/MobileTopSidebar.css:50]: ok

File: src/components/ProtectedRoute.jsx [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 const ProtectedRoute = ({ children }) => {
   const { myToken } = useAuthContext();
+  const location = useLocation();
 
-  return myToken ? children : <Navigate to="/login" />;
+  return myToken ? (
+    children
+  ) : (
+    <Navigate to="/login" replace={true} state={location.pathname} />
+  );
```
Comment (shrey27) [inline src/components/ProtectedRoute.jsx:13]: this is a nice approach to handle authentication and redirection

**Review by shrey27 — APPROVED** [review]
Overall code looks great
Authentication and redirection is well handled
Suggestions:
- add alert or toast messages for better UX
- Add guest credentials on login page
- add _redirects file as on refresh netlify preview is throwing error

PR is good to merge

**Comment by manish591** [comment]
> Overall code looks great Authentication and redirection is well handled Suggestions:
> 
> * add alert or toast messages for better UX
> * Add guest credentials on login page
> * add _redirects file as on refresh netlify preview is throwing error
> 
> PR is good to merge

I have already added _redirects and toast messages in the other PR. Will add guest login. Thanks for suggestion

**Pull request merged**